R/datatables.R

```diff
@@ -102,6 +102,13 @@ datatable = function(
     getOption('DT.options', list()),
     if (is.function(options)) options() else options
   )
+
+  # https://github.com/rstudio/DT/issues/658
+  # must provide a list to options$buttons.
+  # otherwise, scalar string doesn't work expectedly.
+  if (is.character(options[['buttons']]))
```

Comment:
Is this test necessary? I mean, can we unconditionally convert `options[['buttons']]` to a list via `as.list()`? If it is already a list, it won't hurt.

Reply:
I agree with you.

Reply:
Well, I think it is still necessary because `buttons` can take a boolean value: https://datatables.net/reference/option/buttons

Reply:
Yes, but unfortunately, DT doesn't support a boolean value at the present, due to :

https://github.com/rstudio/DT/blob/c6fd8641d551e6f130f5512f9334c8e329228c70/R/datatables.R#L575-L576